test/base.test.js

```diff
@@ -669,3 +669,33 @@ test('skip the cache if operation is Mutation', async ({ equal, same, teardown }
   equal(skipCount, 0)
   equal(hitCount, 0)
 })
+
+test('using all option as string', async (t) => {
+  const app = fastify()
+  app.register(mercurius)
+  app.register(cache, {
+    all: 'true'
+  })
+
+  await t.rejects(app.ready())
+})
+
+test('using ttl option as string', async (t) => {
+  const app = fastify()
+  app.register(mercurius)
+  app.register(cache, {
+    ttl: '10'
+  })
+
+  await t.rejects(app.ready())
+})
+
+test('using cacheSize option as string', async (t) => {
+  const app = fastify()
+  app.register(mercurius)
+  app.register(cache, {
+    ttl: '1024'
+  })
+
+  await t.rejects(app.ready())
+})
```

Comment:
we need to test that validation throws the right errors, also possibly when we register the plugin, not after the ready() method returns

Reply:
This is expected, it's ok.

Reply:
what is expected? that we don't test that the right errors are thrown? :)

Reply:
Ah I misread your comment.

The error will surface _only_ at `app.ready()` or at `await app.register(cache)` level. You will still have a promise to deal with. Validating the error is important.